shop/src/main/java/ro/msg/learning/shop/controller/CategoryController.java

```diff
@@ -0,0 +1,37 @@
+package ro.msg.learning.shop.controller;
+
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.DTO.ProductCategoryDTO;
+import ro.msg.learning.shop.DTO.builders.ProductCategoryBuilder;
+import ro.msg.learning.shop.service.CategoryService;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class CategoryController {
+    private final CategoryService productCategoryService;
+    private final ProductCategoryBuilder productCategoryBuilder;
```

Comment:
The controller should be let's call it a **dumb layer**. All that he should know is authentication (in the future) and who to pass the responsability to (the service in our case). Move the builder logic in the service